var/spack/repos/builtin/packages/geopm/package.py

```diff
@@ -0,0 +1,133 @@
+    variant('procfs', default=True, description='Enable procfs (disable for OSes not using procfs).')
+    variant('mpi', default=True, description='Enable MPI dependent components.')
+    variant('fortran', default=True, description='Build fortran interface.')
+    # TODO: add explicit ruby-ronn dependency. For +doc.
+    # This is currently done by the install process and should be handled
+    # as configure option --with-ronn=/path/to/ronn.
+    # Changes are required in the GEOPM install! To be done in future version.
+    variant('doc', default=True, description='Create man pages with ruby-ronn.')
+    variant('openmp', default=True, description='Build with OpenMP.')
+    variant('ompt', default=False, description='Use OpenMP Tools Interface.')
+    variant('hwloc', default=True, description='Build with hwloc.')
+    variant('gnu-ld', default=False, description='Assume C compiler uses gnu-ld.')
+
+    # Added dependencies.
+    depends_on('m4', type='build')
+    depends_on('automake', type='build')
+    depends_on('autoconf', type='build')
+    depends_on('libtool', type='build')
+    depends_on('numactl', type=('build', 'run'))
+    depends_on('mpi', when='+mpi', type=('build', 'run'))
+    depends_on('hwloc', type=('build', 'run'))
+    depends_on('py-pandas', type='run')
+    depends_on('py-numpy', type='run')
+    depends_on('py-natsort', type='run')
+    depends_on('py-matplotlib', type='run')
+
+    parallel = False
+
+    def install(self, spec, prefix):
+        spec = self.spec
+        args = []
+
+        args.append('--prefix=%s' % prefix)
+
+        if '+debug' in spec:
+            args.append('--enable-debug')
+
+        if '+coverage' in spec:
+            args.append('--enable-coverage')
+
+        if '+overhead' in spec:
+            args.append('--enable-overhead')
+
+        if '~procfs' in spec:
+            args.append('--disable-procfs')
+
+        if '~mpi' in spec:
+            args.append('--disable-mpi')
+        else:
+            args.extend([
+                '--with-mpi-bin={0}'.format(spec['mpi'].prefix.bin),
+                '--with-mpicc={0}'.format(spec['mpi'].mpicc),
+                '--with-mpicxx={0}'.format(spec['mpi'].mpicxx),
+                '--with-mpifc={0}'.format(spec['mpi'].mpifc),
+                '--with-mpif77={0}'.format(spec['mpi'].mpif77),
+            ])
+
+        if '~fortran' in spec:
+            args.append('--disable-fortran')
+
+        if '~doc' in spec:
+            args.append('--disable-doc')
+
+        if '~openmp' in spec:
+            args.append('--disable-openmp')
+
+        if '+ompt' in spec:
+            args.append('--enable-ompt')
+
+        if '+gnu-ld' in spec:
+            args.append('--with-gnu-ld')
+
+        if '+hwloc' in spec:
+            args.append('--with-hwloc={0}'.format(spec['hwloc'].prefix))
+        # else:
+        #     args.append('--without-hwloc')
```

Comment:
I generally prefer these kinds of things if the build script is non-deterministic. Meaning, if the build enables hwloc if it can find it and disables it otherwise, it is better for Spack to explicitly have control over this. We don't want someone building `geopm~hwloc` and getting `--with-hwloc` just because they happened to have it installed on their system.

Reply:
I agree, but my rationale here was the following: 
Having --with-hwloc=/path/to/hwloc explicitly sets it.
Not having that uses the system installed.
But only --without-hwloc enforces not to include hwloc.

Since spack should not use system installed versions, but only ones that explicitly are controlled by spack the uncommented "else --without-hwloc" was already the spack way to do this?
Just to clarify ;)

Reply:
> Not having that uses the system installed.

that's exactly what we want to avoid by explicitly doing `--without-hwloc` when `~hwloc`.